src/main/java/com/company/services/TwitterService.java

```
@@ -35,34 +31,42 @@ public static TwitterService getInstance() {
         return instance;
     }
 
-    public Response postTweet(String message, TwitterAppConfigurationKeys keys) {
+    public TwitterPost postTweet(String message, TwitterAppConfigurationKeys keys) {
+        Twitter twitter = buildTwitter(keys);
         if (StringUtils.isAllBlank(message) || message.length() > 280) {
-            return Response.status(Response.Status.NOT_FOUND).entity(new TwitterErrorResponse(Response.Status.NOT_FOUND.getStatusCode(), "Message parameter cannot be null, empty white spaces, or longer than 280 characters.")).build();
+            logger.error("Form Parameter 'message' cannot be null, empty white spaces, or longer than 280 characters.");
         }
-        Twitter twitter = buildTwitter(keys);
         try {
             Status status = twitter.updateStatus(message);
-            return Response.ok(new TweetResponse(status.getText())).build();
+            return new TwitterPost(new User(status.getUser().getScreenName(), status.getUser().getName(), status.getUser().getProfileImageURL()), status.getText(), status.getCreatedAt());
         } catch (TwitterException e) {
             logger.error("In postTweet: There was an error interacting with the Twitter API and/or Twitter Keys.", e);
-            return Response.status(Response.Status.NOT_FOUND).entity(new TwitterErrorResponse(Response.Status.NOT_FOUND.getStatusCode(), "There was an error when trying to post your tweet.")).build();
+            return null;
```

instead of returning null, you should throw an exception